hydpy/models/rconc/rconc_model.py

```diff
@@ -0,0 +1,288 @@
+        outuh(2.0)
+
+        We solve the underlying ordinary differential equation via the explicit Euler
+        method.  Nevertheless, defining arbitrarily high storage coefficients does not
+        pose any stability problems due to truncating too high outflow values:
+
+        >>> control.recstep(1)
+        >>> derived.dt.update()
+        >>> nmbstorages(5)
+        >>> derived.ksc(inf)
+        >>> model.calc_outuh_sc_v1()
+        >>> fluxes.outuh
+        outuh(2.0)
+
+        Increasing the number of internal calculation steps via parameter |RecStep|
+        results in higher numerical accuracy without violating the water balance:
+
+        >>> derived.ksc(2.0)
+        >>> states.sc = 0.0
+        >>> model.calc_outuh_sc_v1()
+        >>> fluxes.outuh
+        outuh(2.0)
+        >>> states.sc
+        sc(0.0, 0.0, 0.0, 0.0, 0.0)
+
+        >>> control.recstep(10)
+        >>> derived.dt.update()
+        >>> states.sc = 0.0
+        >>> model.calc_outuh_sc_v1()
+        >>> fluxes.outuh
+        outuh(0.084262)
+        >>> states.sc
+        sc(0.714101, 0.542302, 0.353323, 0.202141, 0.103872)
+        >>> from hydpy import round_
+        >>> round_(fluxes.outuh + sum(states.sc))
+        2.0
+
+        >>> control.recstep(100)
+        >>> derived.dt.update()
+        >>> states.sc = 0.0
+        >>> model.calc_outuh_sc_v1()
+        >>> fluxes.outuh
+        outuh(0.026159)
+        >>> states.sc
+        sc(0.850033, 0.590099, 0.327565, 0.149042, 0.057103)
+        >>> round_(fluxes.outuh + sum(states.sc))
+        2.0
+    """
+
+    CONTROLPARAMETERS = (rconc_control.NmbStorages, rconc_control.RecStep)
+    DERIVEDPARAMETERS = (rconc_derived.DT, rconc_derived.KSC)
+    REQUIREDSEQUENCES = (rconc_fluxes.InUH,)
+    UPDATEDSEQUENCES = (rconc_states.SC,)
+    RESULTSEQUENCES = (rconc_fluxes.OutUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> None:
+        con = model.parameters.control.fastaccess
+        der = model.parameters.derived.fastaccess
+        flu = model.sequences.fluxes.fastaccess
+        sta = model.sequences.states.fastaccess
+        if (con.nmbstorages == 0) or modelutils.isinf(der.ksc):
+            flu.outuh = flu.inuh
+        else:
+            flu.outuh = 0.0
+            for _ in range(con.recstep):
+                sta.sc[0] += der.dt * flu.inuh
+                for j in range(con.nmbstorages - 1):
+                    d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                    sta.sc[j] -= d_q
+                    sta.sc[j + 1] += d_q
+                j = con.nmbstorages - 1
+                d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                sta.sc[j] -= d_q
+                flu.outuh += d_q
+
+
+class Set_RConcInput_V1(modeltools.Method):
+    """Set the input for the calculation of the runoff concentration."""
+
+    RESULTSEQUENCES = (rconc_fluxes.InUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model, v: float) -> float:
```

Comment:
Wrong return type.  This is the other reason why the build currently fails.  (Mypy helps you find such bugs.)

Reply:
Thanks for finding this bug. What can be the reason that mypy did NOT find this bug?

Reply:
It definitely should.  Here is the output on my system (Mypy 1.9.0):

```
mypy .\hydpy\
hydpy\models\hland_v4.py:1043: error: Argument 1 to "get_rconcmodel_waterbalance" of "Model" has incompatible type "dict[str, dict[str, float | ndarray[Any, dtype[floating[Any]]]]]"; expected "dict[str, dict[str, dict[str, float | n
darray[Any, dtype[floating[Any]]]]]]"  [arg-type]
hydpy\models\hland_v3.py:1047: error: Argument 1 to "get_rconcmodel_waterbalance" of "Model" has incompatible type "dict[str, dict[str, float | ndarray[Any, dtype[floating[Any]]]]]"; expected "dict[str, dict[str, dict[str, float | n
darray[Any, dtype[floating[Any]]]]]]"  [arg-type]
hydpy\models\hland_v1.py:1653: error: Argument 1 to "get_rconcmodel_waterbalance" of "Model" has incompatible type "dict[str, dict[str, float | ndarray[Any, dtype[floating[Any]]]]]"; expected "dict[str, dict[str, dict[str, float | n
darray[Any, dtype[floating[Any]]]]]]"  [arg-type]
hydpy\models\rconc\rconc_model.py:213: error: Missing return statement  [return]
Found 4 errors in 4 files (checked 411 source files)
```